_posts/2020-05-07-removing-capybara-webkit.md

```diff
@@ -0,0 +1,133 @@
+---
+layout: post
+title: "Removing capybara-webkit"
+date: 2020-04-15 12:00:00
+categories: ["rails", "upgrades"]
+author: arielj
+---
+
+We all know testing is important. We have our unit tests and integration tests to make sure everything is working.
+
+In Ombulabs, we use [capybara](https://github.com/teamcapybara/capybara) for our integration tests so we can interact with the app as a real user would.
+
+Capybara by default uses `rack-test` as the driver, but `rack-test` does not support Javascript so to test some things we need a web browser with Javascript capabilities. Having the browser show up while the tests are running is not that practical, now we can use the headless feature on browsers like Chrome or Firefox (even Edge), but a few years ago the available options were mainly PhantomJS or capybara-webkit (through QtWebKit). Those options were deprecated in favor of the major browsers' headless mode.
+
+This is the process we used to replaced `capybara-webit` gem in a legacy project with a more modern approach using the `webdrivers` gem and headless browser.
+
+<!--more-->
+
+## Issues With Capybara-webkit
+
+- Development was officially suspended on March. [commit](https://github.com/thoughtbot/capybara-webkit/commit/f429d668568ff7349f5e23a085df7fcf1c431fa7#diff-04c6e90faac2675aa89e2176d2eec7d8)
+- Depends on QT so it requires the developer to install extra libraries on the system adding an extra step to make a project run locally, inside docker, or in our CI system.
+- It's difficult to update the webkit engine that's run in the background so you can't have all the new features you would have on the actual browsers
+
+## Process
+
+We will devide this change into 4 different steps:
+1. Replace `capybara-webkit`
+1. Make sure specs passes locally
```

Comment:
Isn't locally and docker the same in this case?

Reply:
Hmmm I meant just running `rake spec` using your OS ruby version and database and everything instead of inside the Docker container, maybe `natively`?